build-and-deploy-cloud-functions.sh

```diff
@@ -0,0 +1,33 @@
+
+PROJECT_ID=$1
+STUDY_MANAGER_SCHEMA=$3
+STUDY_SERVER_SCHEMA=$4
+
+echo "Will deploy to ${PROJECT_ID} using schemas ${STUDY_MANAGER_SCHEMA} and ${STUDY_SERVER_SCHEMA}"
+
+# mvn -Pcloud-function -DskipTests clean install package
+
+
+#echo "Deploying kit dispatcher to ${PROJECT_ID}"
+#gcloud --project=${PROJECT_ID} functions deploy \
+#    tbos-kit-tracking-dispatcher \
+#    --entry-point=org.broadinstitute.dsm.cf.TestBostonKitTrackerDispatcher \
+#    --runtime=java11 \
+#    --trigger-topic=kit-report \
+#    --source=target/deployment \
+#    --set-env-vars="PROJECT_ID=${PROJECT_ID},SECRET_ID=cloud-functions,STUDY_MANAGER_SCHEMA=${STUDY_MANAGER_SCHEMA},STUDY_SERVER_SCHEMA=${STUDY_SERVER_SCHEMA}" \
+#    --egress-settings=all \
+#    --vpc-connector=appengine-default-connect
+
+
+
+echo "Deploying covid order registrar to ${PROJECT_ID}"
+gcloud --project=${PROJECT_ID} functions deploy \
+    order-in-care-evolve \
+    --entry-point=org.broadinstitute.dsm.cf.Covid19OrderRegistrarFunction \
+    --runtime=java11 \
+    --trigger-topic=tbos-ce-orders \
+    --source=target/deployment \
+    --memory=1024MB \
+    --set-env-vars="PROJECT_ID=${PROJECT_ID},SECRET_ID=study-manager-config" \
+    --vpc-connector=projects/broad-ddp-dev/locations/us-central1/connectors/appengine-default-connect
```

Comment:
Looking into the tutorial, I can't totally figure out if we have to deploy them every time ?

Reply:
You'll need to do a deploy anything you want to change something, whether that's code, memory, env vars, or other settings. I've been working with Cloud Functions myself for some Pepper stuff and I have a similar `deploy.sh` script for those functions. Happy to chat to discuss my learnings.